File: JS/6/example6.js
#!/usr/bin/env node

'use strict';

var fs = require('fs');
var readline = require('readline');

var outfile = "out.txt";

var text = "aldsfkjlsadfkj\nalsdfkjalsdfjk\nalsdkjflasdfjk\nlasdkjlsadkjf\n";

console.log("Writing to out file ...");

fs.writeFile(outfile, text, (err) => {
    if (err) throw err;
});

console.log("Reading from out file ...");

const rl = readline.createInterface({
    input: fs.createReadStream(outfile)
});

rl.on('line', (line) => {
    console.log('Line from file:', line);
});

console.log("Putting more text into outfile.");
var more_text = "alsdkjfalfdkj\nalsdfjldfkj\n";

fs.appendFile(outfile, more_text, (err) => {
    if (err) throw err;
});

console.log("Reading entire file content.");
function readWrap(callback) {
    fs.readFile(outfile, 'utf8', function (err, data) {
        if (err) return console.log(err);
        callback(null, data);
    });
}

readWrap(function (err, data) {
    console.log(data);
});

File: Java/basic/src/main/java/org/prodigius/javabasic/example6/Example6.java
package org.prodigius.javabasic.example6;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.StandardOpenOption;
import java.util.stream.Stream;

public class Example6
{
    public static void main(String[] args) throws IOException
    {
        String outfile = "out.txt";
        String text = "aldsfkjlsadfkj\nalsdfkjalsdfjk\nalsdkjflasdfjk\nlasdkjlsadkjf\n";

        System.out.println("Writing to outfile ....");
        Files.write(Paths.get(outfile), text.getBytes());

        System.out.println("Reading outfile line by line ...");

        try (Stream<String> stream = Files.lines(Paths.get(outfile)))
        {
            stream.forEach(System.out::println);
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }

        String moreText = "alsdkjfalfdkj\nalsdfjldfkj\n";

        System.out.println("Adding more to outfile ...");

        try
        {
            Files.write(Paths.get(outfile), moreText.getBytes(), StandardOpenOption.APPEND);
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }

        System.out.println("Reading entire file ....");
        String out = new String(Files.readAllBytes(Paths.get(outfile)));
        System.out.println(out);

    }
}